fix: Dashboard overflow issue if there is banner

iamsivin opened the pull request:
# Pull Request Template

## Description

This PR includes,
1. Fix the top right green action button's inconsistent behavior when a banner is present.
2. Removed `padding` and used `gap` inside the button.
3. Moved some CSS from `layout.scss` file to inline.
4. Minor chat list header spacing fix.

Fixes https://linear.app/chatwoot/issue/CW-2401/top-right-button-inconsitent-behavior-when-a-banner-is-present

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

**Screencast**
https://www.loom.com/share/8da98de15991475d98eab023213ba513?sid=131ed791-5c94-4607-a0dd-55dba3d88285

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

scmmishra commented on app/javascript/dashboard/components/ui/Banner.vue:
```diff
@@ -96,7 +96,7 @@ export default {
 <style lang="scss" scoped>
 .banner {
-  @apply flex items-center gap-4 text-white dark:text-white text-xs py-3 px-4 justify-center;
+  @apply flex items-center h-12 gap-4 text-white dark:text-white text-xs py-3 px-4 justify-center;
```
Yep, just the base styles, the rest can stay since it's out of the scope of this PR. You can make those changes in a separate PR

iamsivin replied:
Done

iamsivin commented on app/javascript/dashboard/assets/scss/_layout.scss:
```diff
@@ -9,87 +9,25 @@ body {
-  flex-grow: 0;
-  min-height: 0;
-  width: 100%;
+  @apply h-full flex-grow-0 min-h-0 w-full;
```
@scmmishra Do you mean moving this `h-full flex-grow-0 min-h-0 w-full` to inline one?

iamsivin replied:
@scmmishra At the moment, we're using the `app-wrapper` class in the super admin. Just want to let you know that if we move it, it might cause some issues.

scmmishra replied:
We can keep the style, just move the top-level (base) style for now, the rest can be migrated later

So the new class list for the component would be `app-wrapper h-full flex-grow-0 min-h-0 w-full`

iamsivin replied:
Done

iamsivin commented on app/javascript/dashboard/routes/dashboard/settings/macros/MacroEditor.vue:
```diff
@@ -141,9 +141,9 @@ export default {
   height: 100vh;
 }
-</style>
+</style> -->
```
Done

scmmishra commented on app/javascript/dashboard/assets/scss/_layout.scss:
```diff
@@ -9,87 +9,25 @@ body {
-  width: 100%;
+  @apply h-full flex-grow-0 min-h-0 w-full;
+
+  .button--fixed-top {
+    @apply fixed ltr:right-2 rtl:left-2 top-2 flex flex-row;
+  }
```
Ideally, for the misaligned button, we should not be using fixed positioning. Instead, it should be enclosed in the parent container with style flex, and aligned using flex alignment properties, see how it is done in the resolve-actions in the conversation view

iamsivin replied:
@scmmishra For the resolve-actions, I think it's straightforward. In this case. There are multiple files with different actions for the button. Currently, the way we have done here is there is a Wrapper component that includes a setting header and different settings route view. And all the button actions are done in different components. In that case, we need to change the logic that we did.

scmmishra replied:
Ah, understood, a solution is to solve this using portals, but that seems like an overkill. We can let this go for now then, and take a look at it at a later point. Feel free to resolve this comment

iamsivin replied:
Sure

nithindavid commented:
@iamsivin, could you let us know which parts of the app need to be tested as this is a big PR?

iamsivin commented:
> @iamsivin, could you let us know which parts of the app need to be tested as this is a big PR?

@nithindavid I have added a loom video in the description. 

Test cases
1. Check whether the dashboard and help center view gets overflow when a banner is present. And the right green button in the settings section. 
2. Uses gap instead of padding in woot-button styles to work well in RTL view. So i removed the padding styles from the button content class. Check whether any button styles are broken or not. 
3. Chat header spacing fix(the search and sidebar toggle button).
4. Cleaned up the layout.scss file and moved some styles that are not using and some to inline.

scmmishra commented:
Tested locally, looks good to go
